File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/decorator/uielement/BackgroundDecorator.java
// BackgroundDecorator.java

package com.bigboxcode.designpattern.decorator.uielement;

public class BackgroundDecorator extends UIDecorator {
    protected BackgroundDecorator(UIElement uiElement) {
        super(uiElement);
    }

    @Override
    public void draw() {
        // Can perform any additional task anywhere in the method

        super.draw();

        // Write code to add background to the element

        System.out.println("Adding Background to the element");
    }
}

File: design-patterns/ts/decorator/ui-element/background-decorator.ts
// background-decorator.ts

import UIDecorator from "./ui-decorator";
import UIElement from "./ui-element";


class BackgroundDecorator extends UIDecorator {
    constructor(uiElement: UIElement) {
        super(uiElement);
    }

    draw(): void {
        super.draw();
        console.log("Adding Background to the element");
    }
}

export default BackgroundDecorator;